### PR TITLE
[DBEX] refactor 0781 failure email logging & monitoring

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

- *This work is behind a feature toggle (flipper): ~YES~/NO*
- *(Summarize the changes that have been made to the platform)*
  - Removes unnecessary memoization and methods in `Form0781DocumentUploadFailureEmail` (from https://github.com/department-of-veterans-affairs/vets-api/pull/19081) based on feedback received in https://github.com/department-of-veterans-affairs/vets-api/pull/19097
- *(If bug, how to reproduce)* N/A
- *(What is the solution, why is this the solution?)*
  - Logs messages identified as failures, which are made known to the user via email notification from VA Notify
  - Increments metrics to capture an issue
  - Supports logging and metrics requirements set by VA stakeholders
- *(Which team do you work for, does your team own the maintenance of this component?)*
  - Disability Benefits Experience Team 2 (dBeX Carbs 🥖), which owns maintenance of all files in this PR
- *(If introducing a flipper, what is the success criteria being targeted?)* N/A

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/95092
- https://github.com/department-of-veterans-affairs/vets-api/pull/19081

## Testing done

- [x] *New code is covered by unit tests*

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
- Impacts logging and metrics data recorded in Datadog

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by NB28VT — APPROVED** [review]
Ah good call making these changes here as well. This looks good to me and the tests pass so I think this is good to go!

File: app/sidekiq/evss/disability_compensation_form/form0781_document_upload_failure_email.rb [hunk 1]
```diff
@@ -87,60 +87,41 @@ def retryable_error_handler(error)
...
-      end
-
-      def zsf_monitor
-        ZeroSilentFailures::Monitor.new(Form526Submission::ZSF_DD_TAG_SERVICE)
-      end
-
-      def call_location
-        cl = caller_locations.first
-        ZeroSilentFailures::Monitor::CallLocation.new(ZSF_DD_TAG_FUNCTION, cl.path, cl.lineno)
-      end
```
Comment (rmtolmach) [inline app/sidekiq/evss/disability_compensation_form/form0781_document_upload_failure_email.rb:143]: What's the reason for moving these out of their own objects and into `log_mailer_dispatch`?

Comment (freeheeling) [reply app/sidekiq/evss/disability_compensation_form/form0781_document_upload_failure_email.rb:143]: Enhanced code structure through increased encapsulation to improve clarity, assigning variables closer to where they're used in an effort to more easily reference relevant data. Happy to have your perspective.

Comment (rmtolmach) [reply app/sidekiq/evss/disability_compensation_form/form0781_document_upload_failure_email.rb:143]: I like keeping methods scoped to one purpose, but I won't block this if that's the style you want to use.

**Review by rmtolmach — APPROVED** [review]
Tag me if you do change anything else and I'll re-review!

**Pull request merged**